kythe/java/com/google/devtools/kythe/analyzers/java/AutoValuePlugin.java

```diff
@@ -0,0 +1,320 @@
+                      method.sym.getSimpleName().toString(),
+                      GeneratedSymbol.builder()
+                          .setAbstractSym(abstractSym.get())
+                          .setGeneratedSym(method.sym)
+                          .build());
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    if (getters.isEmpty()) {
+      return Optional.empty();
+    }
+
+    // Getters must either be all prefixed or not prefixed; trim prefixes, if found.
+    if (getters.keySet().stream().allMatch(AutoValuePlugin::hasGetPropertyPrefix)) {
+      getters = trimPropertyNames(getters);
+    }
+    // Separately, setters must either be all prefixed or not prefixed; trim prefixes, if found.
+    if (setters.keySet().stream().allMatch(s -> s.startsWith("set"))) {
+      setters = trimPropertyNames(setters);
+    }
+
+    // Pair getters and setters by property name.
+    for (Map.Entry<String, GeneratedSymbol> getter : getters.entrySet()) {
+      String name = getter.getKey();
+      Property.Builder prop = Property.builder().setName(name).setGetter(getter.getValue());
+      if (setters.containsKey(name)) {
+        prop.setSetter(setters.get(name));
+      }
+      autoValue.addProperty(prop.build());
+    }
+
+    return Optional.of(autoValue.build());
+  }
+
+  private Optional<Symbol> findAbstractMethod(ClassSymbol abstractClass, MethodSymbol genMethod) {
+    Scope scope = abstractClass.members();
+    for (Symbol abstractSym : scope.getSymbolsByName(genMethod.name)) {
+      if (abstractSym != null
+          && Flags.asModifierSet(abstractSym.flags()).contains(Modifier.ABSTRACT)
+          && genMethod.overrides(abstractSym, abstractClass, types, true)) {
+        return Optional.of(abstractSym);
+      }
+    }
+    return Optional.empty();
+  }
+
+  private static Optional<JCAnnotation> getAnnotation(String name, List<JCAnnotation> annotations) {
+    return annotations.stream().filter(ann -> annotationName(ann).equals(name)).findFirst();
+  }
+
+  private static String annotationName(JCAnnotation ann) {
+    return ann.getAnnotationType().type.tsym.toString();
+  }
+
+  private static Object annotationLiteralValue(JCAnnotation ann, String name) {
+    for (JCExpression expr : ann.getArguments()) {
+      JCAssign a = (JCAssign) expr;
+      if (a.lhs.toString().equals(name) && a.rhs instanceof JCLiteral) {
+        return ((JCLiteral) a.rhs).value;
+      }
+    }
+    return null;
+  }
+
+  private static Map<String, GeneratedSymbol> trimPropertyNames(Map<String, GeneratedSymbol> syms) {
+    ImmutableMap.Builder<String, GeneratedSymbol> unprefixed = ImmutableMap.builder();
+    for (Map.Entry<String, GeneratedSymbol> sym : syms.entrySet()) {
+      unprefixed.put(propertyName(sym.getKey()), sym.getValue());
+    }
+    return unprefixed.build();
+  }
+
+  private static boolean hasGetPropertyPrefix(String name) {
+    return name.startsWith("get") || name.startsWith("is");
```

Comment:
Probably `isPrefixedGetter`, and needs to check that the remainder is CamelCase.